packages/dds/migration-shim/src/test/sharedTreeBug.spec.ts

```diff
@@ -0,0 +1,167 @@
+	ForestType,
+} from "@fluid-experimental/tree2";
+import { type IFluidHandle } from "@fluidframework/core-interfaces";
+import { type IContainerRuntimeOptions } from "@fluidframework/container-runtime";
+import { LoaderHeader } from "@fluidframework/container-definitions";
+
+const treeKey = "treeKey";
+
+class TestDataObject extends DataObject {
+	public async getTree(): Promise<ISharedTree> {
+		const handle: IFluidHandle<IChannel> | undefined =
+			this.root.get<IFluidHandle<IChannel>>(treeKey);
+		assert(handle !== undefined, "No handle found");
+		return (await handle.get()) as ISharedTree;
+	}
+
+	public createTree(type: string): ISharedTree {
+		const channel = this.runtime.createChannel(treeKey, type);
+		this.root.set(treeKey, channel.handle);
+		return channel as ISharedTree;
+	}
+}
+
+const builder = new SchemaBuilder({ scope: "test" });
+const someType = builder.object("abc", {
+	quantity: builder.number,
+});
+
+const schema = builder.intoSchema(SchemaBuilder.required(someType));
+
+function getNewTreeView(tree: ISharedTree): ISharedTreeView {
+	return tree.schematizeView({
+		initialTree: {
+			quantity: 0,
+		},
+		allowedSchemaModifications: AllowedUpdateType.None,
+		schema,
+	});
+}
+
+const testValue = 5;
+
+describeNoCompat("SharedTree", (getTestObjectProvider) => {
+	// Allow us to control summaries
+	const runtimeOptions: IContainerRuntimeOptions = {
+		summaryOptions: {
+			summaryConfigOverrides: {
+				state: "disabled",
+			},
+		},
+	};
+
+	// Registry
+	const sharedTreeFactory = new SharedTreeFactory({
+		jsonValidator: typeboxValidator,
+		forest: ForestType.Reference,
+	});
+
+	const dataObjectFactory = new DataObjectFactory(
+		"TestDataObject",
+		TestDataObject,
+		[sharedTreeFactory],
+		{},
+	);
+
+	// The 2nd runtime factory, V2 of the code
+	const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore({
+		defaultFactory: dataObjectFactory,
+		registryEntries: [dataObjectFactory.registryEntry],
+		runtimeOptions,
+	});
+
+	let provider: ITestObjectProvider;
+
+	beforeEach(async () => {
+		provider = getTestObjectProvider();
+	});
+
+	it("Can create and retrieve tree", async () => {
+		// Setup containers
+		const container1 = await provider.createContainer(runtimeFactory);
+		const testObj1 = (await container1.getEntryPoint()) as TestDataObject;
+		const tree1 = testObj1.createTree(sharedTreeFactory.type);
+
+		await provider.ensureSynchronized();
+		const container2 = await provider.loadContainer(runtimeFactory);
+		const testObj2 = (await container2.getEntryPoint()) as TestDataObject;
+		const tree2 = await testObj2.getTree();
+
+		// Get the views and root object to modify
+		const view1 = getNewTreeView(tree1);
+		const view2 = getNewTreeView(tree2);
+		const treeNode1 = view1.root as unknown as Typed<typeof someType>;
+		const treeNode2 = view2.root as unknown as Typed<typeof someType>;
+
+		await provider.ensureSynchronized();
+
+		// Modify the tree
+		treeNode1.quantity = testValue;
```

Comment:
I think only the proxy-based API is allowed to set values using `=`?

```suggestion
		const treeNode1 = view1.root2(schema);
		const treeNode2 = view2.root2(schema);

		await provider.ensureSynchronized();

		// Modify the tree
		treeNode1.quantity = testValue;
```

Reply:
I run into 0x77f with the other change